### PR TITLE
Adding navigation button to Search Input.

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: #2387 

The previous PR was made against master and github got confused and closed it when I rebased. I'm reopening on behalf of @nadavoran with some tweaks and fixes to pass CI.

### Summary of Changes

* Add the up/down arrow navigation to the SearchInput and use in file search (@nadavoran's original work)
* Tweak the  original PR to pass lint/CI, add the navigation to project search

### Test Plan

- [x] Command-P opens the project search
- [x] type a search with multiple results, click the next/prev arrows to traverse
- [x] type a search with one result and see no arrows
- [x] try the same with file search (both with symbol searching and code searching.)

### Screenshots/Videos

screenshot_20170327_021426
screenshot_20170327_021440
search-nav-demo

### CODE REVIEW

File: src/components/shared/SearchInput.js [hunk 1]
```diff
@@ -39,6 +41,27 @@ const SearchInput = React.createClass({
...
+      { className: "search-nav-buttons" },
+      Svg("arrow-down", {
+        className: classnames("nav-btn", "next"),
+        onClick: handleNext,
+        title: "Next Result",
+      }),
+      Svg("arrow-up", {
+        className: classnames("nav-btn", "prev"),
+        onClick: handlePrev,
+        title: "Previous Result",
```
Comment (wldcordeiro) [inline src/components/shared/SearchInput.js]: Could put these in L10N.

File: src/components/Editor/SearchBar.js [hunk 2]
```diff
@@ -354,6 +354,38 @@ const SearchBar = React.createClass({
...
+          resultCount - 1,
+          selectedResultIndex + 1
+        );
+
+        if (selectedResultIndex === resultCount - 1) {
+          nextResultIndex = 0;
+        }
+        this.setState({ selectedResultIndex: nextResultIndex });
+        this.onSelectResultItem(searchResults[nextResultIndex]);
+      }
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: Hmm  - I'm a bit confused by this logic. 

It looks very similar to what we're doing in search search to determine the next index. Is there an easy way we can unify it?

File: src/components/Editor/SearchBar.js [hunk 3]
```diff
@@ -417,28 +449,35 @@ const SearchBar = React.createClass({
...
-      );
-      this.setState({ selectedResultIndex });
-      this.onSelectResultItem(searchResults[selectedResultIndex]);
+      let nextResultIndex = Math.min(resultCount - 1, selectedResultIndex + 1);
+
+      if (selectedResultIndex === resultCount - 1) {
+        nextResultIndex = 0;
+      }
+      this.setState({ selectedResultIndex: nextResultIndex });
+      this.onSelectResultItem(searchResults[nextResultIndex]);
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: This looks like it can be shared too...

File: src/components/shared/SearchInput.js [hunk 4]
```diff
@@ -39,6 +41,27 @@ const SearchInput = React.createClass({
...
+      })
+    );
+  },
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: SearchInput could be a good component to snapshot test:

https://facebook.github.io/jest/docs/snapshot-testing.html#content

File: src/components/shared/SearchInput.js [hunk 5]
```diff
@@ -39,6 +41,27 @@ const SearchInput = React.createClass({
...
+      return;
+    }
+
+    return dom.div(
+      { className: "search-nav-buttons" },
+      Svg("arrow-down", {
+        className: classnames("nav-btn", "next"),
+        onClick: handleNext,
+        title: "Next Result",
+      }),
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: what's our preference with wrapping - next at the end and previous at  0?

**Comment by jasonLaster** [comment]
This looks much better, I'll review soon!

**Comment by jasonLaster** [comment]
Here's a GIF to see how great it is:

**Comment by wldcordeiro** [comment]
Working on  a rebase/polish commit now. Hopefully will push today (vacation is slowing me down 😆)

**Comment by jasonLaster** [comment]
Hehe. Thanks. I was so close to getting this in yesterday!

File: src/components/Editor/SearchBar.js [hunk 6]
```diff
@@ -640,6 +666,10 @@ const SearchBar = React.createClass({
         onChange: this.onChange,
         onKeyUp: this.onKeyUp,
         onKeyDown: this.onKeyDown,
+        handleNext: e =>
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: hmm, not sure if we need the ternary. if we're not rendering the buttons will these be used?

File: src/components/shared/SearchInput.js [hunk 7]
```diff
@@ -23,6 +23,27 @@ class SearchInput extends Component {
     return Svg("magnifying-glass");
   }
 
+  renderNav() {
+    const { count, handleNext, handlePrev } = this.props;
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: lets put the feature flag here

**Comment by nadavoran** [comment]
Wow that looks great, thanks for adding this and all your work to make it so nice

File: src/components/shared/SearchInput.js (hunk 7 above)
Comment (wldcordeiro) [inline src/components/shared/SearchInput.js]: Oh yeah, forgot to feature flag it. 😆

**Comment by jasonLaster** [comment]
Thanks @nadavoran for getting this started!
Thanks @wldcordeiro!

**Pull request merged**